tedana/workflows/tedana.py

```diff
@@ -680,67 +657,70 @@ def tedana_workflow(
-
-            n_bold_comps = comptable[comptable.classification == "accepted"].shape[0]
-            if (n_restarts < maxrestart) and (n_bold_comps == 0):
+            ica_selection = selection.automatic_selection(comptable, n_echos, n_vols, tree=tree)
+            n_LikelyBOLD_comps = ica_selection.n_LikelyBOLD_comps
+            if (n_restarts < maxrestart) and (n_LikelyBOLD_comps == 0):
                 LGR.warning("No BOLD components found. Re-attempting ICA.")
-            elif n_bold_comps == 0:
+            elif n_LikelyBOLD_comps == 0:
                 LGR.warning("No BOLD components found, but maximum number of restarts reached.")
                 keep_restarting = False
             else:
                 keep_restarting = False
 
+            # If we're going to restart, temporarily allow force overwrite
+            if keep_restarting:
+                io_generator.force = True
             RepLGR.disabled = True  # Disable the report to avoid duplicate text
         RepLGR.disabled = False  # Re-enable the report after the while loop is escaped
+        io_generator.force = force  # Re-enable original overwrite behavior
     else:
         LGR.info("Using supplied mixing matrix from ICA")
         mixing_file = io_generator.get_name("ICA mixing tsv")
         mmix = pd.read_table(mixing_file).values
 
-        if ctab is None:
-            required_metrics = [
-                "kappa",
-                "rho",
-                "countnoise",
-                "countsigFT2",
-                "countsigFS0",
-                "dice_FT2",
-                "dice_FS0",
-                "signal-noise_t",
-                "variance explained",
-                "normalized variance explained",
-                "d_table_score",
-            ]
-            comptable = metrics.collect.generate_metrics(
-                catd,
-                data_oc,
-                mmix,
-                masksum_clf,
-                tes,
-                io_generator,
-                "ICA",
-                metrics=required_metrics,
-            )
-            comptable, metric_metadata = selection.kundu_selection_v2(comptable, n_echos, n_vols)
-        else:
-            LGR.info("Using supplied component table for classification")
-            comptable = pd.read_table(ctab)
-            # Change rationale value of rows with NaN to empty strings
-            comptable.loc[comptable.rationale.isna(), "rationale"] = ""
-
-            if manacc is not None:
-                comptable, metric_metadata = selection.manual_selection(comptable, acc=manacc)
+        required_metrics = [
+            "kappa",
+            "rho",
+            "countnoise",
+            "countsigFT2",
+            "countsigFS0",
+            "dice_FT2",
+            "dice_FS0",
+            "signal-noise_t",
+            "variance explained",
+            "normalized variance explained",
+            "d_table_score",
+        ]
```

Comment:
Not yet. We can rewrite tedana.py to take any tree as an input and only calculate the required metrics listed in that tree, but we decided that should be a separate PR. To do it well, it would be best to make that change as part of a refactored tedana.py #920. Everything is set up to make that change possible, but we messing with enough of the code base already that we wanted to keep tedana.py relatively stable (We removed functionality from tedana.py but we didn't change the remaining functionality)